rust_snuba/src/processors/eap_items.rs

```diff
@@ -170,19 +172,18 @@ impl From<FromSpanMessage> for EAPItem {
                 }
             }
 
-            let mut sampling_factor = 1.0;
             if let Some(measurements) = from.measurements {
                 for (k, v) in measurements {
                     match k.as_str() {
-                        "client_sample_rate" if v.value > 0.0 => sampling_factor *= v.value,
-                        "server_sample_rate" if v.value > 0.0 => sampling_factor *= v.value,
+                        "client_sample_rate" if v.value > 0.0 => res.sampling_factor *= v.value,
+                        "server_sample_rate" if v.value > 0.0 => res.sampling_factor *= v.value,
```

Comment:
nit: Since we process client sample rate and server sample rate exactly same, they can be merged:

```
match k.as_str() {
    "client_sample_rate" | "server_sample_rate" if v.value > 0.0 => {
        res.sampling_factor *= v.value;
    }
    _ => {
        res.attributes.insert_float(k, v.value);
    }
}
```